src/algebraic_geometry/projective_spectrum/topology.lean

```diff
@@ -124,4 +124,295 @@ lemma subset_zero_locus_iff_le_vanishing_ideal (t : set (projective_spectrum 
 ⟨λ h f k, (mem_vanishing_ideal _ _).mpr (λ x j, (mem_zero_locus _ _ _).mpr (h j) k), λ h,
   λ x j, (mem_zero_locus _ _ _).mpr (le_trans h (λ f h, ((mem_vanishing_ideal _ _).mp h) x j))⟩
 
+variable (𝒜)
+/-- `zero_locus` and `vanishing_ideal` form a galois connection. -/
+lemma gc : @galois_connection
+  (ideal A) (order_dual (set (projective_spectrum 𝒜))) _ _
```

Comment:
Wouldn't it be better to use `homogeneous_ideal` here?

Reply:
But shouldn't those be about homogeneous ideals as well? I think that with graded rings, it's best to stay in the world of homogeneous ideals as much as possible.

Reply:
I agree with you that we should stay with homogeneous ideals as much as possible, but wouldn't it be better to have more general  lemma
```lean
lemma zero_locus_anti_mono_ideal {s t : ideal A} (h : s ≤ t) :
  zero_locus 𝒜 (t : set A) ⊆ zero_locus 𝒜 (s : set A)
```
instead of
```lean
lemma zero_locus_anti_mono_homogeneous_ideal {s t : homogeneous_ideal 𝒜} (h : s ≤ t) :
  zero_locus 𝒜 (t : set A) ⊆ zero_locus 𝒜 (s : set A)
```
since the former immediately implies the later

Reply:
Maybe we want three `gc`s? We have `gc_set`. We can rename the current `gc` to `gc_ideal`. And then have `gc` for the homogeneous version?

Reply:
Let me work on this.

Reply:
I have added `gc_homogeneous_ideal` and some lemmas of the corresponding homogeneous version.

Reply:
Thanks, this looks good to me now.